weather_forecast_retrieval/data/hrrr/file_loader.py

```diff
@@ -0,0 +1,289 @@
+
+import pandas as pd
+import utm
+import xarray as xr
+
+from .config_file import ConfigFile
+from .file_handler import FileHandler
+from .grib_file import GribFile
+from .netcdf_file import NetCdfFile
+
+
+class FileLoader(ConfigFile):
+    """
+    Load data from local HRRR files.
+    Currently supports loading from Grib or NetCDF format.
+    """
+    # Maximum hour that local files will be attempted to be read if a previous
+    # hour could not be found or successfully loaded.
+    MAX_FORECAST_HOUR = 6
+    NEXT_HOUR = timedelta(hours=1)
+
+    def __init__(self, config_file=None, external_logger=None):
+        super().__init__(
+            __name__, config_file=config_file, external_logger=external_logger
+        )
+
+        self.start_date = None
+        self.end_date = None
+
+        self._file_loader = None
+        self.data = None
+        self.force_zone_number = None
+
+    @property
+    def file_loader(self):
+        return self._file_loader
+
+    @file_loader.setter
+    def file_loader(self, value):
+        self._file_loader = value
+
+    @property
+    def file_type(self):
+        if self._file_loader is not None:
+            return self._file_loader.SUFFIX
+        else:
+            return None
+
+    def get_saved_data(self,
+                       start_date, end_date, bbox,
+                       output_dir=None, file_type='grib2',
+                       force_zone_number=None,
+                       forecast_flag=False,
+                       var_keys=None):
+        """
+        Get the saved data from above for a particular time and a particular
+        bounding box.
+
+        Args:
+            start_date:     datetime for the start
+            end_date:       datetime for the end
+            bbox:           list of  [lonmin,latmin,lonmax,latmax]
+            output_dir:     Base path to location of files
+            file_type:      'grib' or 'netcdf', determines how to read the file
+            forecast_flag:  weather or not to get forecast hours
+            force_zone_number: UTM zone number to convert datetime to
+            var_keys:       which keys to grab from smrf variables,
+                            default is var_map
+
+        Returns:
+            List containing dataframe for the metadata for each node point for
+            the desired variables
+        """
+
+        if start_date > end_date:
+            raise ValueError('start_date before end_date')
+
+        self.start_date = start_date
+        self.end_date = end_date
+
+        if file_type == GribFile.SUFFIX:
+            self.file_loader = GribFile()
+            var_map = GribFile.VAR_MAP
+        elif file_type == NetCdfFile.SUFFIX:
+            self.file_loader = NetCdfFile()
+            var_map = NetCdfFile.VAR_MAP
+        else:
+            raise Exception('Unknown file type argument')
+
+        self.file_loader.bbox = bbox
+
+        self.log.info('Getting saved data')
+        if var_map is None:
```

Comment:
It doesn't look like this is tested on if we specify a variable map. What if there were a test for just the air temperature?

Reply:
Good call, will add one